packages/token-metadata/src/TokenMetaUtils.ts

```diff
@@ -48,6 +48,40 @@ export class TokenMetaUtils {
     }
   }
 
+  getMetaByFactory(denom: string): TokenMeta | undefined {
+    const [symbolOrName, creatorAddress] = denom.split('/').reverse()
+    const tokenMeta =
+      this.getMetaByName(symbolOrName) || this.getMetaBySymbol(symbolOrName)
+
+    if (!tokenMeta) {
+      return
+    }
+
+    if (tokenMeta.tokenFactories) {
+      const tokenFactory = tokenMeta.tokenFactories.find(
+        (tokenFactory) => tokenFactory.creator === creatorAddress,
+      )
+
+      if (tokenFactory) {
+        return {
+          ...tokenMeta,
+          tokenType: TokenType.TokenFactory,
+          tokenVerification: TokenVerification.Verified,
+        }
+      }
+    }
+
+    if (tokenMeta.tokenFactory?.creator !== creatorAddress) {
+      return
+    }
+
+    return {
+      ...tokenMeta,
+      tokenType: TokenType.TokenFactory,
+      tokenVerification: TokenVerification.Verified,
+    }
+  }
```

Comment:
Maybe we use this refactor opportunity to simplify the tokenmeta typing:
```
export interface TokenMeta {
  name: string
  logo: string
  symbol: string
  decimals: number
  tokenType: TokenType // primary token type
  tokenVerification?: TokenVerification
  coinGeckoId: string

  ibc?: IbcTokenMeta
  spl?: SplTokenMeta
  cw20?: Cw20TokenMeta
  cw20s?: Cw20TokenMetaWithSource[] // When there are multiple variations of the same CW20 token
  erc20?: Erc20TokenMeta
  evm?: EvmTokenMeta
  // tokenFactory?: NativeTokenFactoryMeta < === remove this
  tokenFactories?: NativeTokenFactoryMeta[] < === use this as the default
}
```

Reply:
Yes, I was thinking the same, but the proper way to do it would be to have:

- One interface which is represented within the list `tokens.ts` which always has multiple `cw20s` and `tokenFactories` 
- One interface which we ship to the products which always has only one `cw20`

Reply:
Would the UI implementation look something like this?
```
// in the store
getters: {
    getToken:(state) => (denomOrSymbol: string) {
        return (state.tokens as tokenMeta[]).find(...) as TokenMetaSingular
    }
}
```


```
// in the component 

token.cw20.decimal
token.tokenFactor.decimal 
```

Reply:
Nope, the `TokenFactory.ts` class would handle this, particularly the `getTokenFromMeta` function which will always return one `cw20` and one `tokenFactory`. I'll do the refactor and push here so you guys can see